### ISSUE TITLE
Ad-Hoc connection loss in MHP3rd with 1.4

### ISSUE THREAD

**Issue by Fell** [hrydgard/ppsspp#9522]
So I just spent hours of narrowing this problem down and I can confirm that it exists in 1.4 and is NOT an error in my configuration.

- I used 1.3 and I could play fine with people on the other side of the world.
- Then I just switched the PPSSPPWindows64.exe to the 1.4 version. Same configuration, same players.
- With 1.4, the connection was first established fine. (Sending and receiving messages in the log, seeing players move)
- After about 30 seconds I get disconnected from the group.

Additional notes:
- Only happens during quests, all gathering hall actions work completely fine.
- This happend every time. I didn't manage to stay connected once with 1.4.
- The log shows a call to `sceNetAdhocPtpClose()` so I guess the game disconnects deliberately.
- I could see in the log that every time this happend I didn't send anything for a few seconds. The log showed 10-20 "receive" messages, but there was nothing sent.
- No error messages in the log at the time of connection loss.
- This also happend to me once when I was using v1.3-152-g923d82c.
- Fast Memory, Multithreading and Real Clock Sync were all disabled.
- I used the real MAC address of my network interface.
- I only tested the "HD Ver" of MHP3rd.
- All ports TCP+UDP were forwarded correctly and firewall was disabled (remember it worked in 1.3)
- Tested on coldbird.net

System information:
- Intel Core i5-5690K @ 3.5 GHz
- AMD R9 290
- 8 GB RAM
- Windows 10 Pro 64bit

I hope this is reproducible. I am happy to provide any additional information needed. I'm even capable of using the VS debugger if someone gives me a hint where/what to look for.

Damn this is a long bug report.

TL;DR: 1.3 works, 1.4 disconnects.

**Comment by tywald** [comment]
Do the other players in your group experience the same? I can't reproduce this on Evolve where I usually play.

On coldbird.net though I tried to join randoms but I disconnect when quest starts with any version. I can't coordinate with the randoms so I have no idea if they still use the alternative speed toggle thing during loading screen(I haven't used it since v0.9.8-890-g1445d5a) or if they have problem with the firewall etc.

**Comment by Fell** [comment]
When looking at the debug log, you can clearly see the difference between "not connecting at all" which is caused by firewall issues, and the problem I've described where the communication works for a few seconds and then disconnects out of nothing.

It's interesting that it works for you on Evolve. Has everyone in your group switched to 1.4 yet? Maybe it just won't work with 1.3 players.

**Comment by tywald** [comment]
Not everybody. Before v1.4 we did try have as many people switching to the 1.3-152 build for the GH is full fix(many are still lazy though and stuck with 1.3 but we could play interchangeably. i.e 1.3-152 with 1.3 people with no problems).

Currently I see no one in the group complaining about 1.4 yet; I will observe for a few weeks and see if people switch back or stick with 1.4

**Comment by Fell** [comment]
I just did some testing with some people from the Coldbird Discord. It seems that I am the only one having this issue, but only with 1.4. I don't get it.

[https://cdn.discordapp.com/attachments/182626568584036353/297891679279775747/unknown.png](url)

**Comment by Fell** [comment]
I've narrowed it down! It's actually caused by b3bb84675981983d82307d9a78cae11ad11615c7.

I've cloned the repo, commented those lines out again, and it worked like a charm.

The only question left is now... why?

**Comment by hrydgard** [comment]
Uh.. ok. Weird, not sure what to make of this..
